framework/include/userobjects/BatchMaterial.h

```diff
@@ -35,6 +35,19 @@ struct GatherMatProp
   }
 };
 
+template <typename T>
+struct GatherMatPropOld
+{
+  typedef T type;
+  typedef const MaterialProperty<T> gather_type;
+
+  template <typename M>
+  static gather_type * getPointer(M & mpi, const MaterialPropertyName & name)
+  {
+    return &mpi.template getMaterialPropertyOld<T>(name);
```

Comment:
After https://github.com/idaholab/moose/pull/24673, `getMaterialProperty` takes a state argument with 0 denoting current, 1 denoting old, etc. So if you wanted to slightly reduce code, you could add add an `unsigned int` template argument to the existing `GatherMatProp` struct

Reply:
And by doing so you'd immediately enable older properties

Reply:
That's a cool enhancement! For now I only enabled GatherMatPropTempl<T, 0> and GatherMatPropTempl<T, 1>. I'll leave the other cases (older state, AD) and their regression tests to whoever needs them.